### ISSUE TITLE
Visual Studio 2015 Update 1 crashing with Code Cracker

### ISSUE THREAD

**Issue by giggio** [dotnet/roslyn#7173]
Originally reported by @Cybermaxs on https://github.com/code-cracker/code-cracker/issues/610#issuecomment-160966346.

Here is the stack trace from his event logs:

````
Message : System.ArgumentException: SyntaxTree ' {{CsCodeRemovedForClarity}}'
not found to remove
Parameter name: syntaxTree
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetSemanticModel(SyntaxTree syntaxTree, Boolean ignoreAccessibility)
   at Microsoft.CodeAnalysis.CSharp.CSharpCompilation.CommonGetSemanticModel(SyntaxTree syntaxTree, Boolean ignoreAccessibility)
   at Microsoft.CodeAnalysis.Diagnostics.SuppressMessageAttributeState.IsDiagnosticSuppressed(String id, Location location, SuppressMessageInfo& info)
   at Microsoft.CodeAnalysis.Diagnostics.SuppressMessageAttributeState.ApplySourceSuppressions(Diagnostic diagnostic, Compilation compilation, ISymbol symbolOpt)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.FilterDiagnosticsSuppressedInSource(ImmutableArray`1 diagnostics, Compilation compilation)
   at Microsoft.CodeAnalysis.Diagnostics.AnalysisResult.StoreAnalysisResult(AnalysisScope analysisScope, AnalyzerDriver driver, Compilation compilation)
   at Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.<ComputeAnalyzerDiagnosticsCoreAsync>d__54.MoveNext()
Pile :
   à System.Environment.FailFast(System.String, System.Exception)
   à Microsoft.CodeAnalysis.FailFast.OnFatalException(System.Exception)
   à Microsoft.CodeAnalysis.FatalError.Report(System.Exception, System.Action`1<System.Exception>)
   à Microsoft.CodeAnalysis.FatalError.ReportUnlessCanceled(System.Exception)
   à Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<ComputeAnalyzerDiagnosticsCoreAsync>d__54.MoveNext()
   à Microsoft.CodeAnalysis.CSharp.CSharpCompilation.GetSemanticModel(Microsoft.CodeAnalysis.SyntaxTree, Boolean)
   à Microsoft.CodeAnalysis.CSharp.CSharpCompilation.CommonGetSemanticModel(Microsoft.CodeAnalysis.SyntaxTree, Boolean)
   à Microsoft.CodeAnalysis.Diagnostics.SuppressMessageAttributeState.IsDiagnosticSuppressed(System.String, Microsoft.CodeAnalysis.Location, Microsoft.CodeAnalysis.Diagnostics.SuppressMessageInfo ByRef)
   à Microsoft.CodeAnalysis.Diagnostics.SuppressMessageAttributeState.ApplySourceSuppressions(Microsoft.CodeAnalysis.Diagnostic, Microsoft.CodeAnalysis.Compilation, Microsoft.CodeAnalysis.ISymbol)
   à Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver.FilterDiagnosticsSuppressedInSource(System.Collections.Immutable.ImmutableArray`1<Microsoft.CodeAnalysis.Diagnostic>, Microsoft.CodeAnalysis.Compilation)
   à Microsoft.CodeAnalysis.Diagnostics.AnalysisResult.StoreAnalysisResult(Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver, Microsoft.CodeAnalysis.Compilation)
   à Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<ComputeAnalyzerDiagnosticsCoreAsync>d__54.MoveNext()
   à System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<ComputeAnalyzerDiagnosticsCoreAsync>d__54, Microsoft.CodeAnalysis, Version=1.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]](<ComputeAnalyzerDiagnosticsCoreAsync>d__54 ByRef)
   à Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers.ComputeAnalyzerDiagnosticsCoreAsync(Microsoft.CodeAnalysis.Diagnostics.AnalyzerDriver, Microsoft.CodeAnalysis.Diagnostics.AsyncQueue`1<Microsoft.CodeAnalysis.Diagnostics.CompilationEvent>, Microsoft.CodeAnalysis.Diagnostics.AnalysisScope, System.Threading.CancellationToken)
   à Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<>c__DisplayClass51_2+<<ComputeAnalyzerDiagnosticsAsync>b__1>d.MoveNext()
   à System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<>c__DisplayClass51_2+<<ComputeAnalyzerDiagnosticsAsync>b__1>d, Microsoft.CodeAnalysis, Version=1.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]](<<ComputeAnalyzerDiagnosticsAsync>b__1>d ByRef)
   à Microsoft.CodeAnalysis.Diagnostics.CompilationWithAnalyzers+<>c__DisplayClass51_2.<ComputeAnalyzerDiagnosticsAsync>b__1()
   à System.Threading.Tasks.Task`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].InnerInvoke()
   à System.Threading.Tasks.Task.Execute()
   à System.Threading.Tasks.Task.ExecutionContextCallback(System.Object)
   à System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   à System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   à System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   à System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   à System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   à System.Threading.ThreadPoolWorkQueue.Dispatch()
   à System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
````

**Comment by Cybermaxs** [comment]
Here is simple project to reproduce this issue : https://github.com/Cybermaxs/analyzerbug-repro
I have this crash on VS 2015 only when [Code Craker] (https://github.com/code-cracker/code-cracker)  is activivated ; but the call stack is not related to code craker itself. Several messages are reported in my events logs.

It seems to be related to RedisCacheProvider.cs and RedisCacheProviderTests.cs : if I remove the test file, there is no issue.

*Note : this project  can be seen as a major truncation of an existing project, it is not useful as is but it's enough to reproduce this issue*

**Comment by mavasani** [comment]
Thanks @Cybermaxs and @giggio 

I was able to repro the crash with the provided repro project. There seem to be couple of issues here:

1. Code cracker analyzer [CC0052](https://github.com/code-cracker/code-cracker/blob/master/src/CSharp/CodeCracker/Usage/ReadonlyFieldAnalyzer.cs) is reporting diagnostics across compilations. For your repro, it seems to report a diagnostic with location in the project `AnalyzerBug`, while analyzing a test file in unit test project `AnalyzerBug.Tests`. See screenshot at the end.
2. Analyzer driver should guard against analyzer diagnostics reported with invalid locations and should never crash VS. I will prepare a PR to address this part.

![corrupt_diagnostic](https://cloud.githubusercontent.com/assets/10605811/11544257/c7f477e2-98f5-11e5-9258-8153e6c4546d.png)

**Comment by Cybermaxs** [comment]
Great ! Thanks for your quick reply

**Comment by giggio** [comment]
We will work on the fix on CodeCracker on our end and release it as soon as it is done, which should be soon. Anyone following the problem can check it on https://github.com/code-cracker/code-cracker/issues/610.

**Comment by giggio** [comment]
Ok, the bug on CodeCracker is already fixed on a PR. We will probably release this tomorrow.

**Comment by giggio** [comment]
CodeCracker 1.0.0-rc5 was just released to the Extension Gallery and to Nuget: https://github.com/code-cracker/code-cracker/releases/tag/v1.0.0-rc5

**Comment by mavasani** [comment]
Thanks @giggio. I tried the new package on the repro project, and VS doesn't crash anymore.

Meanwhile, I am working on a fix on our side as well.

**Comment by Cybermaxs** [comment]
Confirmed. fixed with v1.0.0-rc5
